content/tutorial-deep-learning-on-mnist.ipynb

```diff
@@ -0,0 +1,1017 @@
+{
```

Comment:
<p>It would be really nice to close with a plot showing that your NN can identify an image that was not in the training set (I know the prints also show this, but just for a visual element).</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/numpy/numpy-tutorials/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='68'/>

Reply:
Great idea @melissawm Also - I should pick a better loss function 🙃 The error rate it too high for this simple example, but if there's a will, there's a way!

A matplotlib visualization of the error rate falling and accuracy improving (hopefully) over time would also be super — and it's a common practice. Working on it!

Reply:
I'm not 100% sure how to do "inference" with a trained model to classify images, since I don't think we have the [Saved Model](https://www.tensorflow.org/guide/saved_model) or [checkpoint saving](https://www.tensorflow.org/guide/checkpoint) capability in the simple example. But we can do the following:

TODO: 

- [ ] 1. We can add graphs with `matplotlib` for: 
  - The error rate (loss) for training and test sets (y) per epochs (x) (it should be gradually going down).
  - The accuracy rate for training and test sets (y) per epochs (x) (it should be gradually going up).
    - Need to add a few lines to save the loss and accuracy (prediction) data at each epoch by appending them separate lists (for storage).
- [ ] 2. Add a simple image to help new users:
  - Showing a loop: inputs as vectors -> forward pass through layers -> output (prediction) + loss calculation -> backpropagation with gradient descent -> weight update.

Other improvement suggestions:
- [ ] Use MSE or some simple loss function formula instead of the current one.
- [ ] Try Fashion MNIST - a similar dataset but more fun.

Reply:
@melissawm PTAL 😃 Thank you!

Reply:
Fyi @melissawm I'm reusing the main image (a diagram with arrows for forward/back prop + all the details) again in _Model architecture and training summary_ - the section is quite verbose and a picture can be worth a 1000 words 📸

Reply:
Is this 

> @melissawm Can you please double check the code and comments, just in case +1

All good!